src/write_influxdb_udp.c

```diff
@@ -365,136 +365,131 @@ static int wifxudp_escape_string(char *buffer, size_t buffer_size,
     status = snprintf(buffer + offset, buffer_len - offset, __VA_ARGS__);      \
     if ((status < 0) || (status >= (buffer_len - offset))) {                   \
-      sfree(rates);                                                            \
-      return -1;                                                               \
+      return metric_offset;                                                    \
     }                                                                          \
     offset += status;                                                          \
   } while (0)
 
-  BUFFER_ADD_ESCAPE(vl->plugin);
-  BUFFER_ADD(",host=");
-  BUFFER_ADD_ESCAPE(vl->host);
-  if (strcmp(vl->plugin_instance, "") != 0) {
-    BUFFER_ADD(",instance=");
-    BUFFER_ADD_ESCAPE(vl->plugin_instance);
-  }
-  if (strcmp(vl->type, "") != 0) {
-    BUFFER_ADD(",type=");
-    BUFFER_ADD_ESCAPE(vl->type);
-  }
-  if (strcmp(vl->type_instance, "") != 0) {
-    BUFFER_ADD(",type_instance=");
-    BUFFER_ADD_ESCAPE(vl->type_instance);
-  }
+  for (; *idx < fam->metric.num; (*idx)++) {
+    int offset = metric_offset;
+    have_values = false;
+    BUFFER_ADD_ESCAPE(fam->name);
+    metric_t metric = fam->metric.ptr[*idx];
 
-  BUFFER_ADD(" ");
-  for (size_t i = 0; i < ds->ds_num; i++) {
-    if ((ds->ds[i].type != DS_TYPE_COUNTER) &&
-        (ds->ds[i].type != DS_TYPE_GAUGE) &&
-        (ds->ds[i].type != DS_TYPE_DERIVE)) {
-      sfree(rates);
-      return -1;
+    for (size_t j = 0; j < metric.label.num; j++) {
+      label_pair_t label = metric.label.ptr[j];
+      if (strcmp(label.name, "") != 0 && strcmp(label.value, "") != 0) {
+        BUFFER_ADD(",");
+        BUFFER_ADD_ESCAPE(label.name);
+        BUFFER_ADD("=");
+        BUFFER_ADD_ESCAPE(label.value);
+      }
     }
+    BUFFER_ADD(" ");
 
-    if (ds->ds[i].type == DS_TYPE_GAUGE) {
-      if (isnan(vl->values[i].gauge))
-        continue;
-      if (have_values)
-        BUFFER_ADD(",");
-      BUFFER_ADD("%s=%lf", ds->ds[i].name, vl->values[i].gauge);
-      have_values = true;
+    if (fam->type == METRIC_TYPE_GAUGE) {
+      if (!isnan(metric.value.gauge)) {
+        BUFFER_ADD("value=%lf", metric.value.gauge);
+        have_values = true;
+      }
     } else if (wifxudp_config_store_rates) {
-      if (rates == NULL)
-        rates = uc_get_rate_vl(ds, vl);
-      if (rates == NULL) {
+      gauge_t rate;
+      int ret = uc_get_rate(&metric, &rate);
+      if (ret != 0) {
+        WARNING("write_influxdb_udp plugin: "
+                "uc_get_rate failed.");
+        continue;
+      }
+      if (!isnan(rate)) {
+        BUFFER_ADD("value=%lf", rate);
+        have_values = true;
+      }
+    } else {
+      switch (fam->type) {
+      case METRIC_TYPE_COUNTER:
+        BUFFER_ADD("value=%" PRIi64 "i", metric.value.counter);
+        have_values = true;
+        break;
+      case METRIC_TYPE_UNTYPED: /* derive ¿? */
```

Comment:
No, basically a gauge. I'm debating with myself whether we should remove `METRIC_TYPE_UNTYPED`, it seems like a Prometheus legacy that we don't need to introduce.

Reply:
OK, expecting no gauges I guess.

Reply:
Correct, `METRIC_TYPE_COUNTER` is taking the role of derives.